### PR TITLE
feat(go): write an answer the way each shell reads it

### PR DESCRIPTION
**Pull request by jdx**
`RenderAnswer` turns candidates into the protocol a shell's completion machinery expects. Stacked on #984.

| shell | shape |
| --- | --- |
| bash | `value` |
| fish, nu, PowerShell | `value\tdescription` |
| zsh | `value\tdescription\tinsert` |

zsh takes a third field because what it *displays* and what it *types* are not always the same string. Checked against what `usage complete-word --shell zsh` already emits:

```
use^IInstalls a tool and adds the version to mise.toml.^Iuse$
```

## Three rules that look like details and are not

**Descriptions are all-or-nothing per answer.** A column that appears on some rows and not others reads as missing data rather than as an absent description.

**A description is collapsed onto one line, not truncated at the first break**, so a two-line description still says both halves. The protocols are line-based and a break would look like another candidate. This is why the one-lining *moved here* from `describe` in #984, where it was throwing the second half away — `Candidates` now carries the whole text and the renderer decides how to fit it.

**"Paths belong here too" is a whole line**, not a flag on the protocol, because every one of the five shells can already split output into lines and look at the last one. `\x01` opens it because no candidate can contain a control character.

## Naming

`RenderAnswer` rather than `Render`, which already belongs to failures. Two things in one package turning a value into text for a terminal is reason enough to say which.

## Still to do

Running the `complete` scripts a spec can declare, which means subprocesses — a decision for the layer above this one.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Additive completion-output formatting with broad unit tests; no changes to parsing, binding, or security-sensitive paths.
> 
> **Overview**
> Adds **`argv.RenderAnswer`** so completion **`Answer`** values serialize to the line protocol each shell expects: bash values only; fish, nu, and PowerShell use `value` + tab + description; zsh adds a third insert field with **`zshQuote`** for safe typing.
> 
> Rendering enforces **all-or-nothing description columns** (only among candidates that actually emit), **collapses multi-line help with `oneLine`** instead of truncating, **drops candidates whose values contain control chars/tabs/newlines** rather than repairing them, and appends **`\x01files` / `\x01dirs` marker lines** when paths should complete too.
> 
> **`describe` in `complete.go`** now attaches the full short help (including newlines); one-lining moved here from candidate building. README documents **`RenderAnswer`**; tests cover per-shell shapes and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f37999a9ad5ca932d042f4887c2ebd8617ae6981. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: go/argv/complete_shell.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		switch shell {
+		case Bash:
+			out.WriteString(c.Value)
+		case Zsh:
+			// Display, then description, then what to type: a candidate containing
+			// a space or a quote has to reach the command line intact.
+			out.WriteString(c.Value + "\t" + description + "\t" + zshQuote(c.Value))
+		default:
+			out.WriteString(c.Value)
+			if described {
```
[the remark this answers is not in the document]
Comment (jdx) [reply go/argv/complete_shell.go:106]: Right. Both protocols are lines with tab-separated fields, so zsh in particular would have taken the text after an embedded tab as the description and the text after that as what to *insert* — the shell putting something on the command line that nobody offered.

Values go through the same one-lining descriptions already did. The test asserts the field and row counts each shell expects rather than the text, since that is what the protocol actually is.

A candidate normally comes from a spec and contains neither, but a `complete` script can produce anything, and rearranging the protocol is a worse failure than a missing candidate.

_This comment was generated by Claude Code._

File: go/argv/complete_shell.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		}
+	}
+
+	for _, c := range a.Candidates {
+		// The protocols are lines with tab-separated fields, so a value carrying
+		// either would be read as more rows or more fields. A candidate normally
+		// comes from a spec and contains neither, but a `complete` script can
+		// produce anything, and a completion that rearranges the protocol is a
+		// worse failure than a missing candidate.
+		value := oneLine(c.Value)
```
[the remark this answers is not in the document]
Comment (jdx) [reply go/argv/complete_shell.go]: You are right, and the comment three lines above the bug already said so: "a completion that rearranges the protocol is a worse failure than a missing candidate". Collapsing the delimiter took neither option — it kept the candidate *and* changed it, so the shell would report success while the CLI received an argument nobody offered.

Such a candidate is dropped now. The user types the value themselves and it works, which is the honest failure.

Two things beyond the report:

- **Every control character, not the three that delimit.** The marker lines that mean "paths belong here too" open with `\x01`, so a value beginning with one was read as a marker. The comment on `FilesMarker` claims no candidate can contain a control character — that claim is true now instead of aspirational.
- **Descriptions still collapse.** They are prose, nothing is typed from them, and a two-line help should still say both halves on one line. Other control characters are folded out of those as well: a description is displayed by the shell, and an escape sequence displayed is an escape sequence run.

_This comment was generated by Claude Code._

File: go/argv/complete_shell.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	// Descriptions are all-or-nothing per answer: a column that appears on some
+	// rows and not others reads as missing data rather than as an absent
+	// description.
+	described := false
+	for _, c := range a.Candidates {
+		if c.Describe != "" {
+			described = true
+			break
+		}
+	}
```
[the remark this answers is not in the document]
Comment (jdx) [reply go/argv/complete_shell.go:76]: Right — the flag was computed over every candidate, including the ones dropped a few lines below, so a description on a dropped row turned the column on for the survivors and left an empty field on each. That is the all-or-nothing rule broken by the answer it was deciding for.

It is decided over the rows that will actually be written now, and the case has a test: a dropped row carrying the only description leaves `plain\n` for fish, nu and PowerShell rather than `plain\t\n`.

_This comment was generated by Claude Code._
